jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/editor/CloudWatchLogStream.kt

```diff
@@ -33,107 +24,94 @@ import software.aws.toolkits.jetbrains.services.cloudwatch.logs.actions.TailLogs
-            arrayOf(LogStreamDateColumn(), LogStreamMessageColumn()),
-            mutableListOf<OutputLogEvent>(),
-            // Don't sort in the model because the requests come sorted
-            -1,
-            SortOrder.UNSORTED
-        )
-        logsTable = TableView(model).apply {
-            setPaintBusy(true)
-            autoscrolls = true
-            emptyText.text = message("loading_resource.loading")
-            tableHeader.reorderingAllowed = false
-        }
-        // TODO fix resizing
-        logsTable.columnModel.getColumn(0).preferredWidth = 150
-        logsTable.columnModel.getColumn(0).maxWidth = 150
-        logsTable.autoResizeMode = JTable.AUTO_RESIZE_LAST_COLUMN
-        logsPanel = ScrollPaneFactory.createScrollPane(logsTable)
-    }
+    private val logStreamTable: LogStreamTable = LogStreamTable(project, logGroup, logStream, LogStreamTable.TableType.LIST)
+    private var searchStreamTable: LogStreamTable? = null
 
     init {
-        logStreamActor = LogStreamActor(project.awsClient(), logsTable, logGroup, logStream)
-        Disposer.register(this, logStreamActor)
+        logsPanel.setContent(logStreamTable.component)
+        Disposer.register(this, logStreamTable)
         searchLabel.text = "${project.activeCredentialProvider().displayName} => ${project.activeRegion().displayName} => $logGroup => $logStream"
         searchField.emptyText.text = message("cloudwatch.logs.filter_logs")
-        logsTable.autoResizeMode = JTable.AUTO_RESIZE_ALL_COLUMNS
-        logsPanel.verticalScrollBar.addAdjustmentListener {
-            if (logsTable.model.rowCount == 0) {
-                return@addAdjustmentListener
-            }
-            if (logsPanel.verticalScrollBar.isAtBottom()) {
-                launch {
-                    // Don't load more if there is a logStreamingJob because then it will just keep loading forever at the bottom
-                    if (logStreamingJob == null) {
-                        logStreamActor.channel.send(LogStreamActor.Messages.LOAD_FORWARD)
-                    }
-                }
-            } else if (logsPanel.verticalScrollBar.isAtTop()) {
-                launch { logStreamActor.channel.send(LogStreamActor.Messages.LOAD_BACKWARD) }
+
+        addAction()
+        addActionToolbar()
+        addSearchListener()
+
+        launch {
+            if (startTime != null && duration != null) {
+                logStreamTable.channel.send(LogStreamActor.Message.LOAD_INITIAL_RANGE(startTime, duration))
+            } else {
+                logStreamTable.channel.send(LogStreamActor.Message.LOAD_INITIAL())
             }
         }
-        launch {
-            try {
-                if (startTime != null && duration != null) {
-                    logStreamActor.loadInitialRange(startTime, duration)
+    }
+
+    private fun addSearchListener() {
+        searchField.addActionListener(object : ActionListener {
+            private var lastText = ""
```

Comment:
i think this is fine as is, but can you confirm that a new instance of `ActionListener` is created for each instance of `CloudWatchLogStream`?

Reply:
Reconfirmed this works as expected